Basic EEH test case for PE

vrbagalkote opened the pull request:
[job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/969729/job.txt)
This script injects one EEH error on all PE device in
the machine

Signed-off-by: Venkat R B <vrbagal1@linux.vnet.ibm.com>

PraveenPenguin commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+        cmd = "cat /sys/kernel/debug/powerpc/eeh_enable"
+        output = process.system_output(cmd, ignore_status=True, shell=True)
+        if output != '0x1':
+            print "EEH is not enabled, please enable via FSP"
```
please use `self.skip()`

PraveenPenguin commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+            % self.max_freeze
+        process.system(cmd, ignore_status=True, shell=True)
+        self.function = str(self.params.get('function')).split(" ")
+        print "===================Testing EEH Frozen PE======================"
```
using self.log.info will be good

PraveenPenguin commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+        cmd = "echo %d > /sys/kernel/debug/powerpc/eeh_max_freezes"\
+            % self.max_freeze
+        process.system(cmd, ignore_status=True, shell=True)
+        self.function = str(self.params.get('function')).split(" ")
```
can this will be any default value ?

vrbagalkote replied:
We have functions from 0-17 so, its better for user to decide which function he wants to inject.

narasimhan-v replied:
Please mention the permissible values in Readme file

PraveenPenguin commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+            self.pci_class_name = pci.get_pci_class_name(self.addr)
+            self.pci_interface = pci.get_interfaces_in_pci_address(
+                self.addr, self.pci_class_name)[-1]
+            print "PCI addr = %s" % self.addr
```
all can be move in self.log.info and put in two lines or single lines on your preference

PraveenPenguin commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+            print "PCI class name = %s" % self.pci_class_name
+            print "PCI interface = %s" % self.pci_interface
+            for func in self.function:
+                print "Running error injection on pe %s function %s"\
```
here as please  sel.log.info

PraveenPenguin commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+                        break
+                    print "PE recovered successfully"
+
+    def run_pe(self, func, pci_class_name, pci_interface, pci_mem_addr, pci_mask):
```
please use some valid name as function name

PraveenPenguin commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+                    continue
+                else:
+                    if not self.check_eeh_pe_recovery(self.addr):
+                        print "PE %s recovery failed after first EEH"\
```
i think it can be fail condition please use `self.fail()`

narasimhan-v commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+        if output != '0x1':
+            print "EEH is not enabled, please enable via FSP"
+            sys.exit(1)
+        self.max_freeze = int(self.params.get('max_Freeze', default='1'))
```
Is int conversion really needed here ?

vrbagalkote replied:
I am planning to enhance the script to inject max_freeze + 1 error, So I need this

narasimhan-v commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+        Test to execute basic error injection on PE
+        """
+        self.pci_address = pci.get_pci_addresses()
+        for self.addr in self.pci_address:
```
`for self.addr in pci.get_pci_addresses()`

narasimhan-v commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+        self.return_code = self.error_inject(func, pci_class_name,
+                                             pci_interface, pci_mem_addr,
+                                             pci_mask)
+        if self.return_code != 0:
```
Will this be good here?
`if self.return_code != EEH_HIT`

narasimhan-v commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+        cmd = "errinjct eeh -v -f %s -s %s/%s -a %s -m %s; echo $?"\
+            % (func, pci_class_name, pci_interface, pci_mem_addr, pci_mask)
+        res = process.system_output(cmd, ignore_status=True, shell=True)
+        print res
```
Is this needed ?

narasimhan-v commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+        """
+        cmd = "dmesg | grep -i 'EEH: Notify device driver to resume'; echo $?"
+        tries = 60
+        for _ in range(1, tries+1):
```
Here and all other places
`for _ in range(0, tries):`

vrbagalkote commented:
@PraveenPenguin  @narasimhan-v , Addressed the comments. I have included scenario, because script can be enhanced to iterate depending upon the eeh_max_freeze count and also number of the function. That I will take as part of enhancement.

PraveenPenguin commented on io/pci/PowerVMEEH.py:
```diff
@@ -0,0 +1,188 @@
+            self.log.info("PCI mem_addr = %s" % self.pci_mem_addr)
+            self.log.info("PCI mask = %s" % self.pci_mask)
+            self.log.info("PCI class name = %s" % self.pci_class_name)
+            self.log.info("PCI interface = %s" % self.pci_interface)
```
why not it can be in single line 

` self.log.info("PCI addr = %s,PCI mem_addr = %s,CI mask = %s" %( self.addr,self.pci_mem_addr,self.pci_mask)`

other three also club in one

vrbagalkote replied:
Sure, I can write everything in one line. But just want to know, what difference does it make. I thought code readability will be good.

PraveenPenguin replied:
i think in other way as if all inter connected value  flash in single line it easy to validate , any way it is your test than take your call as it just a suggestion :)

vrbagalkote replied:
I beg your pardon...But I would like it the way it is now. :)

PraveenPenguin replied:
sure merging thanks for the patch

PraveenPenguin commented:
@narasimhan-v thanks
